### PR TITLE
feat(orchestration,workflows): pipe oncomplete and workflow preparation

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
Add pipe onComplete callback and preparation function to exportWorkflow
It also exposes the current DistributedTransaction instance to the transaction handler

### CODE REVIEW

File: packages/workflows/src/helper/workflow-export.ts [hunk 1]
```diff
@@ -26,7 +26,8 @@ export type WorkflowResult<TResult = unknown> = {
 
 export const exportWorkflow = <TData = unknown, TResult = unknown>(
   workflowId: Workflows,
-  defaultResult?: string
+  defaultResult?: string,
+  dataPreparation?: (data: TData) => Promise<unknown>
```
Comment (olivermrbl) [inline packages/workflows/src/helper/workflow-export.ts:31]: comment: Nice, I like it!

**Review by olivermrbl — APPROVED** [review]
LGMT!

**Comment by olivermrbl** [comment]
@adrien2p, OK to merge?

File: packages/workflows/src/helper/__tests__/workflow-export.spec.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    const prepare = jest.fn().mockImplementation(async (data) => {
+      data.__transformed = true
+      transformedInput = data
+
+      return data
+    })
+
+    let a: LocalWorkflow
+
+    const work = exportWorkflow("id" as any, "result", prepare)
```
Comment (adrien2p) [inline packages/workflows/src/helper/__tests__/workflow-export.spec.ts]: suggestion: change result to result_step

File: packages/workflows/src/helper/__tests__/workflow-export.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  it("should prepare the input data before initializing the transaction", async function () {
+    let transformedInput
+    const prepare = jest.fn().mockImplementation(async (data) => {
+      data.__transformed = true
+      transformedInput = data
+
+      return data
+    })
+
+    let a: LocalWorkflow
```
Comment (adrien2p) [inline packages/workflows/src/helper/__tests__/workflow-export.spec.ts]: todo: not sure it is used

File: packages/workflows/src/helper/__tests__/workflow-export.spec.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    const wfHandler = work()
+
+    const input = {
+      test: "payload",
+    }
+
+    const { result } = await wfHandler.run({
+      input,
+      resultFrom: "result_step",
```
Comment (adrien2p) [inline packages/workflows/src/helper/__tests__/workflow-export.spec.ts]: suggestion: rm

**Review by adrien2p — APPROVED** [review]
LGTM 🚀

**Pull request merged**